docs/source/reference/api-server/api-server-admin-deploy.rst

```diff
@@ -289,10 +175,165 @@ Test the API server by curling the health endpoint:
+
+        * To disable using the hosting Kubernetes cluster, set ``kubernetesCredentials.useApiServerCluster=false`` in the Helm chart values.
+        * To use a different namespace for tasks, set ``kubernetesCredentials.inclusterNamespace=<namespace>`` in the Helm chart values.
+
+        .. tip::
+
+            The default permissions granted to the API server works out of box. For further hardening, you can refer to :ref:`Setting minimum permissions in helm deployment <minimum-permissions-in-helm>` to understand the permissions and how to customize them.
+
+        To authenticate to other clusters, first create a Kubernetes secret with the kubeconfig file with :ref:`necessary permissions <cloud-permissions-kubernetes>`:
+
+        .. code-block:: bash
+
+            kubectl create secret generic kube-credentials \
+              --namespace $NAMESPACE \
+              --from-file=config=~/.kube/config
+
+
+        Once the secret is created, set ``kubernetesCredentials.useKubeconfig=true`` and ``kubernetesCredentials.kubeconfigSecretName`` in the Helm chart values to use the kubeconfig for authentication:
+
+        .. code-block:: bash
+
+            helm upgrade --install skypilot skypilot/skypilot-nightly --devel \
+              --namespace $NAMESPACE \
+              # keep the Helm chart values set in the previous step
+              --reuse-values \
+              --set kubernetesCredentials.useKubeconfig=true \
+              --set kubernetesCredentials.kubeconfigSecretName=kube-credentials
+
+        .. tip::
+
+            If you are using a kubeconfig file that contains `exec-based authentication <https://kubernetes.io/docs/reference/access-authn-authz/authentication/#configuration>`_ (e.g., GKE's default ``gke-gcloud-auth-plugin`` based authentication), you will need to strip the path information from the ``command`` field in the exec configuration.
+            You can use the ``exec_kubeconfig_converter.py`` script to do this.
+
+            .. code-block:: bash
+
+                python -m sky.utils.kubernetes.exec_kubeconfig_converter --input ~/.kube/config --output ~/.kube/config.converted
+
+            Then create the Kubernetes secret with the converted kubeconfig file ``~/.kube/config.converted``.
+
+        To use multiple Kubernetes clusters, you will need to add the context names to ``allowed_contexts`` in the SkyPilot config. An example config file that allows using the hosting Kubernetes cluster and two additional Kubernetes clusters is shown below:
+
+        .. code-block:: yaml
+
+            kubernetes:
+              allowed_contexts:
+              # The hosting Kubernetes cluster, you cannot set if the hosting is disabled by kubernetesCredentials.useApiServerCluster=false
+              - in-cluster
+              # The additional Kubernetes context names in the kubeconfig you configured
+              - context1
+              - context2
```

Comment:
This needs to be a part of the helm command above with the prehook we supported. Otherwise, people may not know how to configure this on the API server.